kernel/src/boot_stage2.rs

```diff
@@ -142,10 +142,13 @@ global_asm!(
 
         /* Determine the PTE C-bit position from the CPUID page. */
 
+        /* Locate the table.  The pointer to the CPUID page is 12 bytes into
+         * the stage2 startup structure. */
+        movl 12(%ebp), %ecx
         /* Read the number of entries. */
-        mov CPUID_PAGE, %eax
+        movl (%ecx), %eax
         /* Create a pointer to the first entry. */
-        leal CPUID_PAGE + 16, %ecx
+        leal 12(%ecx), %ecx
```

Comment:
I guess you meant the following here:
```suggestion
        leal 16(%ecx), %ecx
```

Reply:
Yes - thanks for catching this!  I'm not entirely sure why this fails for you but not for me; I suspect that different hypervisors populate the initial CPUID page differently such that in my case it just happens to work.